### PR TITLE
Prep code for PHP 7.4 release

### PR DESCRIPTION
**Pull request by indytechcook**
## Description

Drupal updates to work with PHP 7.4.  This includes a Drupal core update.  The servers will still be on php 7.1.  This PR makes the code compatible with 7.4 when the servers are upgraded.

## Testing done

- Run yaml-tests
- Edit and save nodes on the site.

QA Steps 
- [x] Visit /admin/reports/status and validate that the page loads.
- [x] Paragraph Validation:
* Edit Node https://va-gov-cms.lndo.site/node/2727/edit
* Update the `WYSIWYG` within the "Main Content" to `Connect with a care coordinator, They are great`

Edit Detail Page Returning service member care | VA gov CMS 2020-03-30 10-35-47

* Save the Node
* Verify the title was updated

VA Wilmington Health Care | Returning service member care 2020-03-30 10-36-19

- [x] Build vets-website, any errors will occur in the build validaiton steps.  Below is the command:

```yarn run watch -- --drupal-address=http://va-gov-cms.lndo.site --drupal-user=axcsd452ksey --pull-drupal```

## Definition of Done
- [ ] Product release notes 
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)

---
This change is [Reviewable](https://reviewable.io/reviews/department-of-veterans-affairs/va.gov-cms/1280)

### CODE REVIEW

**Comment by indytechcook** [comment]
When I uninstall `workflow_participants` it also removed `views.view.my_workflow`.

File: docroot/themes/custom/vagovadmin/gulp-tasks/sass-lint.js [hunk 1]
```diff
@@ -3,21 +3,21 @@
...
       ])
 
       .pipe(plugins.sassLint({
         rules: {
-          // Find sass-lint rules in .sass-lint.yml
+          // Find sass-lint rules in .sass-lint.yml.
         },
         files: {
-          //ignore: 'design-library/**/_typey**.scss'
+          // ignore: 'design-library/**/_typey**.scss'.
```
[the remark this answers is not in the document]
Comment (indytechcook) [reply docroot/themes/custom/vagovadmin/gulp-tasks/sass-lint.js]: This was just `phpcbf` doing it's thing

**Comment by indytechcook** [comment]
This can be merged without upgrading the servers since the code is backwards compatible.  Currently devshop is php 7.3.

**Comment by swirtSJW** [comment]
> This can be merged without upgrading the servers since the code is backwards compatible. Currently devshop is php 7.3.
@indytechcook   Sorry, I'm not quite following this comment.   The pr environment for this is claiming 7.1.33
http://pr1280.ci.cms.va.gov/admin/reports/status
image

So these changes are only impacting local landos and not CI or dev,staging.prod?

**Comment by ElijahLynn** [comment]
To test on BRD:
1. Make sure to push branch to this repo (not fork)
1. Take the SHA and build here http://jenkins.vfs.va.gov/job/builds/job/cms/
1. Then deploy it to DEV here http://jenkins.vfs.va.gov/job/deploys/job/cms-vagov-dev/

File: patches/paragraphs_browser/remove-paragraph-group-3116501-2.patch [hunk 2]
```diff
@@ -0,0 +1,18 @@
+diff --git a/src/Plugin/Field/FieldWidget/InlineParagraphsBrowserWidgetTrait.php b/src/Plugin/Field/FieldWidget/InlineParagraphsBrowserWidgetTrait.php
+index b01aebc3..3fcb6f70 100644
+--- a/src/Plugin/Field/FieldWidget/InlineParagraphsBrowserWidgetTrait.php
++++ b/src/Plugin/Field/FieldWidget/InlineParagraphsBrowserWidgetTrait.php
+@@ -94,7 +94,7 @@ public function formElement(FieldItemListInterface $items, $delta, array $elemen
+       $group = $groups->getGroup($group_name);
+       if ($group) {
+         $element['top']['paragraph_type_title']['info'] = array(
+-          '#markup' => '<span class="paragraph-group">' . $group->getLabel() . '</span> ' . $current_title,
```
Comment (swirtSJW) [inline patches/paragraphs_browser/remove-paragraph-group-3116501-2.patch]: This patch change seems out of scope.  How is this related to updating to php 7.4?

Comment (indytechcook) [reply patches/paragraphs_browser/remove-paragraph-group-3116501-2.patch]: We already had the patch, I had to move it a file since with the other path form 'paragraphs_browser` caused the original patch not to apply.

File: composer.lock [hunk 3]
```diff
@@ -4037,7 +4043,6 @@
                 "patches_applied": {
                     "2815221 - Add quickedit to the latest-revision route": "https://www.drupal.org/files/issues/2019-11-27/2815221-125.patch",
                     "Allow menu items which link to unpublished nodes to be selected in the parent item selector": "https://www.drupal.org/files/issues/drupal-core-allow-menu-items-to-link-to-unpublished-2807629-7-8.3.x.patch",
-                    "Views ui disabling causing wsod on status page.": "https://www.drupal.org/files/issues/2019-12-13/Content_Moderation_requirements_check_relies_on_Views_UI_module-3099364-13.patch",
```
[the remark this answers is not in the document]
Comment (indytechcook) [reply composer.lock]: Yes: https://www.drupal.org/project/drupal/issues/3099364

File: patches/paragraphs_browser/remove-paragraph-group-3116501-2.patch (hunk 2 above)
Comment (swirtSJW) [reply patches/paragraphs_browser/remove-paragraph-group-3116501-2.patch]: Seems odd that we had a patch that did apply from d.o and then had to move it locally.   Do you know the cause?

Comment (indytechcook) [reply patches/paragraphs_browser/remove-paragraph-group-3116501-2.patch]: Because the other patch added (https://www.drupal.org/project/paragraphs_browser/issues/3057621) caused this one to no longer apply.  The other patch is in RTBC status while the status for this issue is needs review: https://www.drupal.org/project/paragraphs_browser/issues/3116501.  This is a modification from the one on D.O to allow it to be applied correctly with the other patch.

**Comment by indytechcook** [comment]
Here is the result of the build script.

```
Delta: -148.42mB
Metalsmith build finished!  Starting webpack-dev-server...
Browserslist: caniuse-lite is outdated. Please run next command `yarn upgrade caniuse-lite browserslist`
Browserslist: caniuse-lite is outdated. Please run next command `yarn upgrade`
ℹ ｢wdm｣: Time: 60975ms
Built at: 03/30/2020 9:59:37 AM
ℹ ｢wdm｣: Compiled successfully.
```

File: patches/paragraphs_browser/remove-paragraph-group-3116501-2.patch (hunk 2 above)
Comment (swirtSJW) [reply patches/paragraphs_browser/remove-paragraph-group-3116501-2.patch]: Ok that makes sense.    That information only exists in your head.  Lets get some shorter version of that into the comment within the patches section of the composer.json so there is some breadcrumb  of that.  Maybe the url of the issue?  Otherwise we have a local patch hanging out with no reference at all back to the actual issue.

Comment (indytechcook) [reply patches/paragraphs_browser/remove-paragraph-group-3116501-2.patch]: We can't have comments in the composer file so I updated the key of the object.

**Comment by ElijahLynn** [comment]
> Here is the result of the build script.
> 
> ```
> Delta: -148.42mB
> Metalsmith build finished!  Starting webpack-dev-server...
> Browserslist: caniuse-lite is outdated. Please run next command `yarn upgrade caniuse-lite browserslist`
> Browserslist: caniuse-lite is outdated. Please run next command `yarn upgrade`
> ℹ ｢wdm｣: Time: 60975ms
> Built at: 03/30/2020 9:59:37 AM
> ℹ ｢wdm｣: Compiled successfully.
> ```

Does this mean it was tested on DEV/STAGING (BRD)?

Do we still have to create a follow up story for CI (DevShop)?

**Comment by indytechcook** [comment]
@ElijahLynn that is the output of running the build against php7.1 which is what is already on devshop/BRD.  I'm working on follow up PRs for PHP 7.4 on the actually DEVSHOP/BRD environment.
